Sources/InfomaniakCore/Extensions/Collection+Identity.swift

```diff
@@ -0,0 +1,40 @@
+/*
+ Infomaniak Core - iOS
+ Copyright (C) 2023 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import SwiftUI
+
+@available(iOS 13.0, *)
+extension Collection where Element: Identifiable<Int> {
+    /// Compute a stable id for the given collection by doing a XOR with the id of each child
+    /// - Discussion: If the id of the children is an int it is used as is, else the `hashValue` of the `Identifiable` object will
+    /// be used
+    func collectionId(baseId: AnyHashable? = nil) -> Int {
+        reduce(baseId?.hashValue ?? 0) { $0 ^ $1.id }
+    }
+}
+
+@available(iOS 13.0, *)
+extension Collection where Element: Identifiable {
+    /// Compute a stable id for the given collection by doing a XOR with the id of each child
+    /// - Discussion: If the id of the children is an int it is used as is, else the `hashValue` of the `Identifiable` object will
+    /// be used
+    func collectionId(baseId: AnyHashable? = nil) -> Int {
```

Comment:
I understand we do not want to make an automatic conformance to `identifiable` for collections, this could go wrong with realm for instance.

But we could have our own protocol, like a `CollectionIdentifiable` for `func collectionId(baseId: AnyHashable? = nil) -> Element` Might be useful to abstract it all.

Reply:
Discuss, indeed it would feel hacky on the API design front and thus not a good idea.